src/core/src/FabricGeometry.cpp

```diff
@@ -290,72 +290,188 @@ void FabricGeometry::setGeometry(
-        const auto g = glm::linearRand(0.0f, 1.0f);
-        const auto b = glm::linearRand(0.0f, 1.0f);
-        displayColorFabric[0] = pxr::GfVec3f(r, g, b);
-    } else {
-        displayColorFabric[0] = DEFAULT_VERTEX_COLOR;
-    }
-
-    displayOpacityFabric[0] = DEFAULT_VERTEX_OPACITY;
-
-    if (hasTexcoords) {
-        const auto& texcoords = hasImagery ? imageryTexcoords : texcoords_0;
-
-        srw.setArrayAttributeSize(_pathFabric, FabricTokens::primvars_st, texcoords.size());
-
-        auto stFabric = srw.getArrayAttributeWr<glm::fvec2>(_pathFabric, FabricTokens::primvars_st);
-
-        texcoords.fill(stFabric);
-    }
-
-    if (hasNormals) {
-        srw.setArrayAttributeSize(_pathFabric, FabricTokens::primvars_normals, normals.size());
-
-        auto normalsFabric = srw.getArrayAttributeWr<glm::fvec3>(_pathFabric, FabricTokens::primvars_normals);
-
-        normals.fill(normalsFabric);
-    }
-
-    if (hasVertexColors) {
-        srw.setArrayAttributeSize(_pathFabric, FabricTokens::primvars_vertexColor, vertexColors.size());
-
+    if (primitive.mode == 0) { //if tile is a point cloud
+
+        const auto numVoxels = positions.size();
+        const float quadHalfSize = 1.5f;
+        srw.setArrayAttributeSize(_pathFabric, FabricTokens::points, static_cast<size_t>(numVoxels * 8));
+        srw.setArrayAttributeSize(_pathFabric, FabricTokens::faceVertexCounts, numVoxels * 2 * 6);
+        srw.setArrayAttributeSize(_pathFabric, FabricTokens::primvars_displayColor, numVoxels * 8);
+
+        srw.setArrayAttributeSize(_pathFabric, FabricTokens::faceVertexIndices, numVoxels * 6 * 2 * 3);
+
+        auto pointsFabric = srw.getArrayAttributeWr<pxr::GfVec3f>(_pathFabric, FabricTokens::points);
+        auto faceVertexCountsFabric = srw.getArrayAttributeWr<int>(_pathFabric, FabricTokens::faceVertexCounts);
+        auto faceVertexIndicesFabric = srw.getArrayAttributeWr<int>(_pathFabric, FabricTokens::faceVertexIndices);
+
+        std::vector<glm::fvec3> vertexColorsData(numVoxels);
+        gsl::span<glm::fvec3> vertexColorsSpan(vertexColorsData);
+        if (hasVertexColors) {
+            vertexColors.fill(vertexColorsSpan);
+            srw.setArrayAttributeSize(_pathFabric, FabricTokens::primvars_vertexColor, numVoxels * 8);
+        }
         auto vertexColorsFabric = srw.getArrayAttributeWr<glm::fvec3>(_pathFabric, FabricTokens::primvars_vertexColor);
 
-        vertexColors.fill(vertexColorsFabric);
+        size_t vertIndex = 0;
+        size_t vertexCountsIndex = 0;
+        size_t faceVertexIndex = 0;
+        size_t voxelCounter = 0;
+        size_t vertexColorsIndex = 0;
+        for (size_t voxelNum = 0; voxelNum < numVoxels; voxelNum++) {
+            auto centerGlm = positions.get(voxelNum);
+            pxr::GfVec3f center{centerGlm.x, centerGlm.y, centerGlm.z};
+
+            pointsFabric[vertIndex++] = pxr::GfVec3f{-quadHalfSize, -quadHalfSize, -quadHalfSize} + center;
```

Comment:
Oh, they're byte-compatible. Great to know. I was careful to only pass glm data types down to kernels because of some runtime errors, but there were overlapping issues. Good to know it's not byte compatibility.